### ISSUE TITLE
Outer assertions should retry

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#543]
Cypress Version: 0.19.4

Making a note here that outer assertions attached to a `.then` do not retry. This should be fixed sometime in `0.20.x`

Both of these fail:

```coffeescript
it "does it retry thens?", ->
  obj = {foo: "bar"}

  cy.wrap(obj).then ->
    setTimeout ->
      obj.foo = "baz"
    , 1000

    return obj
  .should("deep.eq", {foo: "baz"})

it "does it retry shoulds outside then?", ->
  obj = {foo: "bar"}

  cy.wrap(obj).then ->
    setTimeout ->
      obj.foo = "baz"
    , 1000

    return cy.wrap(obj)
  .should("deep.eq", {foo: "baz"})
```

I believe they should both pass.

Moving the `should` inside of the `then` works as expected:

```coffeescript
it "does it retry thens?", ->
  obj = {foo: "bar"}

  cy.wrap(obj).should ->
    setTimeout ->
      obj.foo = "baz"
    , 1000

    expect(obj).to.deep.eq({foo: "baz"})

it "does it retry shoulds outside then?", ->
  obj = {foo: "bar"}

  cy.wrap(obj).then ->
    setTimeout ->
      obj.foo = "baz"
    , 1000

    return cy.wrap(obj).should("deep.eq", {foo: "baz"})
```

**Comment by brian-mann** [comment]
The 2nd example test does pass in `0.20.0` now. Opening a new issue for the first example because this is not a problem with command logic - `cy.then` just needs its own retry logic. If that is implemented it will work now as well.

**Comment by brian-mann** [comment]
The code for this is done, but has yet to be released. We'll update this issue and reference the changelog when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
